MIAUI/MIAUI/Data/TasksRepository.cs

```
@@ -0,0 +1,68 @@
+﻿namespace MIAUI.Data;
+
+public class TasksRepository
+{
+    string _dbPath;
+    public string StatusMessage { get; set; }
+
+    private SQLiteConnection conn;
+
+    private void Init ()
+    {
+        if (conn is not null)
+            return;
+
+        conn = new SQLiteConnection (_dbPath);
+        conn.CreateTable<Model.Task> ();
+    }
+
+    public TasksRepository (string dbPath)
+    {
+        _dbPath = dbPath;
+    }
+    public async void AddNewTask (string taskName)
+    {
+        int result = 0;
+        try
+        {
+            Init ();
+            // basic validation to ensure a name was entered
+            if (string.IsNullOrEmpty (taskName))
+                throw new Exception ("Valid name required");
+
+            result = conn.Insert (new Model.Task(taskName));
+            result = 0;
```

```suggestion
```
This line should be removed as it hides any errors